src/Catalog/CatalogItem.cs

```diff
@@ -10,11 +10,25 @@ namespace NuGet.Services.Metadata.Catalog
 {
     public abstract class CatalogItem
     {
+        Uri _baseAddress;
+
         public DateTime TimeStamp { get; set; }
 
         public Guid CommitId { get; set; }
 
-        public Uri BaseAddress { get; set; }
+        public Uri BaseAddress
+        {
+            get
+            {
+                return AliasesAddress(_baseAddress, CustomDomain);
```

Comment:
every time BaseAddress is called AliasesAddress will be recalculated. To improve perf you can calculate this ones, on _baseAddress set